lib/scholar/manifold/trimap.ex

```diff
@@ -113,6 +113,16 @@ defmodule Scholar.Manifold.Trimap do
       doc: ~S"""
       Metric used to compute the distances.
       """
+    ],
+    algorithm: [
+      type: {:in, [:nndescent, :large_vis]},
```

Comment:
I would add at least `:brute`. Maybe custom k-NN graph construction algorithm to be passed as a module as well.

Reply:
Is there any benefit in using brute other than nndescent or large_vis?

Reply:
I'd say it's the best one to use for smaller datasets.

Reply:
Perhaps making the selection automatic depending on dataset size (sample size and number of features) would be ideal.

Reply:
I will force use brute for `n < 100` and otherwise one of these two approximated algorithms, is it ok or do we want to add `:brute` anyway?

Reply:
Maybe use brute-force for $N \times D^2 \leq T$ for some constant $T$ (e.g. $10^5$ or $10^6$ or so).
I would add `:brute` anyway.

Looking at it now, we might wanna change `predict` to `transform` in these algorithms as well.

Reply:
for $N = 20000$ and $D = 2$, we rather won't use brute, generally, it's more effective to have this condition only on N I guess, I can increase this to 500

Reply:
Sounds good to me.